logs/logs.go

```diff
@@ -16,7 +16,7 @@ var ErrOutputStopped = errors.New("Output plugin stopped")
 
 // A LogCollection is a collection of LogSrc, a plugin which can provide many LogSrc
 type LogCollection interface {
-	FindLogSrc() []LogSrc
+	FindLogSrc(ctx context.Context) []LogSrc
```

Comment:
We should not be modifying this. I do not see a good reason to break this interface by adding a context parameter. Please revert this.

Reply:
Same with below  comment.
```
I'm still doing the benchmark between semaphore and only pushed it for committing a snapshot. Have not done the final PR yet.
```